sway/commands/fullscreen.c

```diff
@@ -0,0 +1,40 @@
+#include <wlr/types/wlr_wl_shell.h>
+#include "log.h"
+#include "sway/commands.h"
+#include "sway/config.h"
+#include "sway/tree/container.h"
+#include "sway/tree/view.h"
+#include "sway/tree/layout.h"
+
+// fullscreen toggle|enable|disable
+struct cmd_results *cmd_fullscreen(int argc, char **argv) {
+	struct cmd_results *error = NULL;
+	if (config->reading) return cmd_results_new(CMD_FAILURE, "fullscreen", "Can't be used in config file.");
+	if (!config->active) return cmd_results_new(CMD_FAILURE, "fullscreen", "Can only be used when sway is running.");
+	if ((error = checkarg(argc, "fullscreen", EXPECTED_AT_LEAST, 1))) {
+		return error;
+	}
+	struct sway_container *container =
+		config->handler_context.current_container;
+	if (container->type != C_VIEW) {
+		return cmd_results_new(CMD_INVALID, "fullscreen",
+				"Only views can fullscreen");
```

Comment:
Containers can be fullscreen too :D

Reply:
That requires replacing the call to `wlr_output_set_fullscreen_surface()` when rendering, and that's not something I want to do at this time.

Reply:
Can you prepare a list of the limitations of your PR so we can track what work remains to be done in a separate ticket?

Reply:
bump

Reply:
* Containers need to be fullscreenable.
* Surfaces can request to be fullscreened on a specific output, which is currently ignored.
* Fullscreen is not frame perfect.

I think that's it.

Reply:
That's kind of a big issue. Would fixing it take a lot of work?

Reply:
It's part of a larger epic, we can keep this patch smaller and more focused and deal with it later.

Reply:
Basically nothing is pixel perfect right now. When resizing a view you'll get background artifacts. When closing a view, same. Fixing all of these requires #1650.